dbms/src/Functions/domain.h

```diff
@@ -9,13 +9,19 @@ namespace DB
 {
 
 /// Extracts host from given url.
+template <bool ignore_scheme = true>
 inline StringRef getURLHost(const char * data, size_t size)
 {
     Pos pos = data;
     Pos end = data + size;
 
     if (end == (pos = find_first_symbols<'/'>(pos, end)))
-        return {};
+    {
+        if (ignore_scheme)
+            pos = data;
```

Comment:
The code became less readable. Better to introduce additional variable `slash_pos`...